src/NzbDrone.Core/MediaFiles/UpdateMovieFileService.cs

```diff
@@ -107,27 +104,9 @@ public void Handle(MovieScannedEvent message)
                 return;
             }
 
-            var movies = _movieService.MoviesWithFiles(message.Movie.Id);
-
-            var movieFiles = new List<MovieFile>();
-            var updated = new List<MovieFile>();
-
-            foreach (var group in movies.GroupBy(e => e.MovieFileId))
-            {
-                var moviesInFile = group.Select(e => e).ToList();
-                var movieFile = moviesInFile.First().MovieFile;
-
-                movieFiles.Add(movieFile);
-
-                if (ChangeFileDate(movieFile, message.Movie))
-                {
-                    updated.Add(movieFile);
-                }
-            }
-
-            if (updated.Any())
+            if (message.Movie.HasFile && ChangeFileDate(message.Movie.MovieFile, message.Movie))
```

Comment:
MovieScannedEvent gets ran right at the end of a scan and the movie object that gets passed thru is from the original scan call. On first import of a file this means tts coming over to UpdateMovieFileService with message.Movie.HasFile = true from the import, however message.Movie.movieFile is null as it was originally pulled from the db when no moviefile exists and thus nothing is joined, so this fails out with a null ref on the initial import where movie is picked up. We should likely go back to query from DB here so that the movie object returned has the moviefile joined correctly.

```
[Error] EventAggregator: UpdateMovieFileService failed while processing [MovieScannedEvent]

[v10.0.0.38878] System.NullReferenceException: Object reference not set to an instance of an object.
   at NzbDrone.Core.MediaFiles.UpdateMovieFileService.ChangeFileDate(MovieFile movieFile, Movie movie) in C:\Users\qstic\Documents\Visual Studio 2017\Projects\radarr-ui\src\NzbDrone.Core\MediaFiles\UpdateMovieFileService.cs:line 43
   at NzbDrone.Core.MediaFiles.UpdateMovieFileService.Handle(MovieScannedEvent message) in C:\Users\qstic\Documents\Visual Studio 2017\Projects\radarr-ui\src\NzbDrone.Core\MediaFiles\UpdateMovieFileService.cs:line 107
   at NzbDrone.Core.Messaging.Events.EventAggregator.PublishEvent[TEvent](TEvent event) in C:\Users\qstic\Documents\Visual Studio 2017\Projects\radarr-ui\src\NzbDrone.Core\Messaging\Events\EventAggregator.cs:line 98
```